src/main/java/io/appform/jsonrules/expressions/equality/NotEqualsExpression.java

```diff
@@ -36,23 +33,41 @@
 @ToString(callSuper = true)
 public class NotEqualsExpression extends JsonPathBasedExpression {
     private Object value;
+	private boolean extractValueFromPath;
 
     public NotEqualsExpression() {
         super(ExpressionType.not_equals);
     }
 
     @Builder
-    public NotEqualsExpression(String path, Object value, PreOperation<?> preoperation) {
-        super(ExpressionType.not_equals, path, preoperation);
+    public NotEqualsExpression(String path, Object value, boolean extractValueFromPath, Boolean defaultResult, PreOperation<?> preoperation) {
+        super(ExpressionType.not_equals, path, ComparisonUtils.getDefaultResult(defaultResult, true), preoperation);
         this.value = value;
+        this.extractValueFromPath = extractValueFromPath;
     }
 
     @Override
     protected boolean evaluate(ExpressionEvaluationContext context, String path, JsonNode evaluatedNode) {
+    	if (extractValueFromPath) {
+			JsonNode jsonNode = context.getNode().at(String.valueOf(value));
```

Comment:
Are you using 4 spaces as tab? If not please change it to that.

Reply:
Done, at all places